src/app/core/services/policy-definition-builder.ts

```diff
@@ -58,24 +58,39 @@ export class PolicyDefinitionBuilder {
   private buildTimePeriodRestrictionPermissions(
     formValue: NewPolicyDialogFormValue,
   ): Permission[] {
-    const start = formValue.range!!.start!!;
-    const end = addDays(formValue.range!!.end!!, 1);
+    if (!formValue.rangeIsOpenEnded) {
+      const start = formValue.range!!.start!!;
+      const end = addDays(formValue.range!!.end!!, 1);
 
-    return [
-      this.policyDefinitionUtils.buildPermission({
-        constraints: [
-          this.policyDefinitionUtils.buildAtomicConstraint(
-            ExpressionLeftSideConstants.PolicyEvaluationTime,
-            'GEQ',
-            start.toISOString()!,
-          ),
-          this.policyDefinitionUtils.buildAtomicConstraint(
-            ExpressionLeftSideConstants.PolicyEvaluationTime,
-            'LT',
-            end.toISOString()!,
-          ),
-        ],
-      }),
-    ];
+      return [
+        this.policyDefinitionUtils.buildPermission({
+          constraints: [
+            this.policyDefinitionUtils.buildAtomicConstraint(
+              ExpressionLeftSideConstants.PolicyEvaluationTime,
+              'GEQ',
+              start.toISOString()!,
+            ),
+            this.policyDefinitionUtils.buildAtomicConstraint(
+              ExpressionLeftSideConstants.PolicyEvaluationTime,
+              'LT',
+              end.toISOString()!,
+            ),
+          ],
+        }),
+      ];
+    } else {
+      const start = formValue.rangeStart!!;
+      return [
+        this.policyDefinitionUtils.buildPermission({
+          constraints: [
+            this.policyDefinitionUtils.buildAtomicConstraint(
+              ExpressionLeftSideConstants.PolicyEvaluationTime,
+              'GEQ',
+              start.toISOString()!,
+            ),
```

Comment:
this is really not necessary, but I think this is a good place to do this:

since you use this code three times, I think you could create a helper method in this class:

```ts
// Idk the exact types, but the buildAtomicConstraint function might be of help
private evaluationTime(op: Operator, value: Date): Constraint {
  // ...
}
```